packages/dashboard/app/withdraw/utils/hiddenTokens.tsx

```
@@ -0,0 +1,117 @@
+                        
+                        localStorage.removeItem(HIDDEN_TOKENS_KEY);
+                        setHiddenTokens([]);
+                        
+                        throw new Error('Hidden tokens data is invalid');
+                    }
+                }
+            } catch (error) {
+                console.error('Error loading hidden tokens from localStorage', error);
+                setError(error as string);
+                setHiddenTokens([]);
+                localStorage.setItem(HIDDEN_TOKENS_KEY, JSON.stringify([]));
+            } finally {
+                setIsProcessing(false);
+            }
+        }
+        loadHiddenTokens();
+    }, []);
+
+    const hideToken = useCallback((address: string) => {
+        if (address === 'ETH') return; // Don't allow hiding ETH
+
+        const lowercaseAddress = address.toLowerCase();
+        try {
+            ethereumAddressSchema.parse(lowercaseAddress);
+        } catch (validationError) {
+            console.error('Invalid address format:', validationError);
+            return;
+        }
+
+        const newHiddenTokens = [...hiddenTokens, lowercaseAddress];
+        setHiddenTokens(newHiddenTokens);
+
+        try {
+            localStorage.setItem(HIDDEN_TOKENS_KEY, JSON.stringify(newHiddenTokens));
+        } catch (error) {
+            console.error('Error saving to localStorage', error);
+        }
+    }, [hiddenTokens]);
+
+    const unhideToken = useCallback((address: string) => {
+        const lowercaseAddress = address.toLowerCase();
+        
+        const newHiddenTokens = hiddenTokens.filter(
+            (tokenAddr: string) => tokenAddr !== lowercaseAddress
+        );
+        setHiddenTokens(newHiddenTokens);
+
+        try {
+            localStorage.setItem(HIDDEN_TOKENS_KEY, JSON.stringify(newHiddenTokens));
+        } catch (error) {
+            console.error('Error saving to localStorage', error);
```

Don't swallow and hide the error from the user